### PR TITLE
ROC-5491 Notify parties when legal order is drawn

### PR DESCRIPTION
**Pull request by simad**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-5491

### Change description ###

Notifying claimant and defendant when a legal order is drawn, using the notify client.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/EmailUtils.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import java.util.Optional;
+
+public final class EmailUtils {
+
+    private EmailUtils() {
+        // utility class, no instances
+    }
+
+    public static Optional<String> getDefendantEmail(Claim claim) {
```
Comment (sabahirfan) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/EmailUtils.java]: Would be nice to have unit tests for this.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java [hunk 2]
```diff
@@ -200,4 +200,18 @@ public static String referenceForDefendant(String claimReferenceNumber) {
...
+    public static class LegalOrderDrawn {
+        private LegalOrderDrawn() {
+            // do not instantiate
+        }
+
+        public static String referenceForClaimant(String claimReferenceNumber) {
+            return reference("to-%s-legal-order-drawn-%s", CLAIMANT, claimReferenceNumber);
+        }
+
+        public static String referenceForDefendant(String claimReferenceNumber) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java]: Both of these method can be replaced with single method with additional parameter for party. Please also add unit tests for this.

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/EmailUtils.java (hunk 1 above)
Comment (simad) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/utils/EmailUtils.java]: oof I was sure I added a test for that. Will do :)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java (hunk 2 above)
Comment (simad) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java]: 👍 was probably following the existing pattern a bit mindlessly, will take a look

Comment (simad) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java]: I was testing it in `OrderDrawnNotificationServiceTest` but having a unit test closer to the code is always good :D added

**Pull request merged**